lib/astutils.cpp

```diff
@@ -1877,6 +1877,16 @@ std::vector<const Token *> getArguments(const Token *ftok)
     return astFlatten(startTok, ",");
 }
 
+int getArgumentPos(const Variable* var, const Function* f)
+{
+    auto arg_it = std::find_if(f->argumentList.begin(), f->argumentList.end(), [&](const Variable& v) {
```

Comment:
I wonder if it would be a good idea to capture var by value not by reference.

Reply:
Why? It seems simpler and faster to capture by ref as it has lexical scoping and avoids creating copies.

Reply:
the motivation would be if we get some const safety. if it doesn't have a performance penalty.. that would be good. for a trivial example I made the assembler output from gcc is identical.

Reply:
The variables are already declared const.

Reply:
I assume that in the lambda you could write `var=nullptr;` and that will happily compile.

Reply:
Yea, it can be modified with `var=nullptr`. I can make the pointer const.